### PR TITLE
chore(typescript): Type :allthethings:

### PR DESCRIPTION
**Pull request by m3fawner**

### CODE REVIEW

File: src/components/EstimatedNetGrant.tsx [hunk 1]
```diff
@@ -12,15 +12,20 @@ import {
...
 import { getDollar, getPercentString } from '../util/formatters';
 import useAlphaVantageAPI from '../hooks/useAlphaVantageAPI';
 import RHFInput from './RHFInput';
 
+type CastedFormData = {
+  symbol: string | undefined,
+  stateTax: number | undefined,
+  ssEarnings: number | undefined,
+  sharesGranted: number | undefined,
+}
```
Comment (moon-tea) [inline src/components/EstimatedNetGrant.tsx]: A more idiomatic approach would probably be to make these optionals. 
```ts
symbol?: string,
```

File: src/components/RSUvOptions/types.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export type dataRow = {
```
Comment (moon-tea) [inline src/components/RSUvOptions/types.ts]: all types should be `PascalCase`.

File: src/components/RHFInput.tsx [hunk 3]
```diff
@@ -8,13 +10,23 @@ import {
...
 import get from 'lodash/get';
 
+type RHFInputProps = {
+    onChange?: ReactEventHandler,
+    onBlur?: ReactEventHandler,
+    name: string,
+    left?: ReactNode,
+    right?: ReactNode,
+    label: ReactNode,
+    errors?: Record<string, { message: string }>,
```
Comment (moon-tea) [inline src/components/RHFInput.tsx]: I see most of these props on `RHFCheckBoxProps`, possible refactor is to find what is common, collect, import, and compose from there.

File: src/components/EstimatedNetGrant.tsx [hunk 4]
```diff
@@ -74,6 +81,18 @@ const HelpItems = () => (
...
+  grantProceeds?: number,
+  federalTaxes?: number,
+  medicareTaxes?: number,
+  ssTaxes?: number,
+  stateTaxes?: number,
+  netProceeds?: number,
+  sharesToCover?: number,
+  ssEarnings?: number,
+  sharesGranted?: number,
+}
```
Comment (moon-tea) [inline src/components/EstimatedNetGrant.tsx]: Not sure if this is helpful, but I thought I would point out:

If everything in a type is going to be optional, you can instead define it as non-optional and then in the implementation later you can use `Partial<CalculatedData>`, which will make each of the fields optional. 

You can also do some cool things like `type SsRelated = Pick<CalculatedData, 'ssTaxes' | 'ssEarnings'>;` to enforce certain things later on. Of course `Omit` is also something you can do. I find `Omit` to be important when making HOCs.

File: src/components/EstimatedNetGrant.tsx (hunk 1 above)
Comment (m3fawner) [reply src/components/EstimatedNetGrant.tsx]: Ah, very good call.

File: src/components/EstimatedNetGrant.tsx (hunk 4 above)
Comment (m3fawner) [reply src/components/EstimatedNetGrant.tsx]: > If everything in a type is going to be optional, you can instead define it as non-optional and then in the implementation later you can use Partial<CalculatedData>, which will make each of the fields optional.

Is this just an "ease of declaration" method?

I do like Pick, Omit, Exclude, etc. It's a cool feature of TS

Comment (m3fawner) [reply src/components/EstimatedNetGrant.tsx]: I think this convo will remain open, but I'm going to merge the pull request. I actually realized I just needed to join the two type defs

**Pull request merged**